### PR TITLE
Remove some Linq usages in ComponentModel.Annotations

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #56631

### CODE REVIEW

**Review by roji — APPROVED** [review]

File: src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EnumDataTypeAttribute.cs [hunk 1]
```diff
@@ -100,7 +99,7 @@ public override bool IsValid(object? value)
         }
 
         private static bool IsEnumTypeInFlagsMode(Type enumType) =>
-            enumType.GetCustomAttributes(typeof(FlagsAttribute), false).Any();
+            enumType.GetCustomAttributes(typeof(FlagsAttribute), false).Length > 0;
```
Comment (stephentoub) [inline src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EnumDataTypeAttribute.cs:102]: Could this be:
```suggestion
            enumType.IsDefined(typeof(FlagsAttribute), false);
```
?

File: src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs [hunk 2]
```diff
@@ -190,6 +188,8 @@ public static class Validator
         public static bool TryValidateValue(object value, ValidationContext validationContext,
             ICollection<ValidationResult>? validationResults, IEnumerable<ValidationAttribute> validationAttributes)
         {
+            ArgumentNullException.ThrowIfNull(validationAttributes, nameof(validationAttributes));
```
Comment (stephentoub) [inline src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs:191]:
```suggestion
            ArgumentNullException.ThrowIfNull(validationAttributes);
```

File: src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs [hunk 3]
```diff
@@ -312,12 +319,14 @@ public static void ValidateObject(object instance, ValidationContext validationC
         public static void ValidateValue(object value, ValidationContext validationContext,
             IEnumerable<ValidationAttribute> validationAttributes)
         {
-            if (validationContext == null)
+            ArgumentNullException.ThrowIfNull(validationContext, nameof(validationContext));
+            ArgumentNullException.ThrowIfNull(validationAttributes, nameof(validationAttributes));
```
Comment (stephentoub) [inline src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs:323]:
```suggestion
            ArgumentNullException.ThrowIfNull(validationContext);
            ArgumentNullException.ThrowIfNull(validationAttributes);
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs (hunk 2 above)
Comment (eerhardt) [reply src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs:191]: I didn't know if this worked yet or not. If it does, I'll change to this.

**Pull request merged**